openwisp_radius/api/freeradius_views.py

```diff
@@ -141,7 +169,24 @@ def authenticate(self, request):
         self.check_organization(request)
         uuid, token = self.get_uuid_token(request)
         if not uuid and not token:
-            return self._radius_token_authenticate(request)
+            if request.data.get('status_type', None) in UNSUPPORTED_STATUS_TYPES:
+                return
+            username = request.data.get('username') or request.query_params.get(
+                'username'
+            )
+            username, request = self._handle_mac_address_authentication(
+                username, request
+            )
+            if username is None and request is None:
+                # When using MAC roaming, the "username" attribute contains the
+                # MAC address (calling_station_id). If there's no open-session
+                # for the given MAC address, then don't authenticate the user.
+                #
+                # NOTE: We return "None" here instead of raising "AuthenticationFailed"
+                # to prevent unnecessary authentication rejected errors in freeradius
+                # logs.
```

Comment:
I'd rewrite it like:

```
# When using MAC auth roaming, the "username" attribute contains the
# MAC address (calling_station_id). When the user connects the first time,
# since it doesn't have any open session, the mac address authorization
# will fail, in order to avoid filling the log with failed mac address
# authorization requests we return "None" here
# (instead of raising "AuthenticationFailed").
# freeradius will take care of rejecting the authorization if no
# explicit "Auth-Type: Accept" is returned
```